management/affinitygroups/client.go

```diff
@@ -0,0 +1,130 @@
+package affinitygroups
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+)
+
+const (
+	azureCreateAffinityGroupURL = "/affinitygroups"
+	azureGetAffinityGroupURL    = "/affinitygroups/%s"
+	azureListAffinityGroupsURL  = "/affinitygroups"
+	azureUpdateAffinityGroupURL = "/affinitygroups/%s"
+	azureDeleteAffinityGroupURL = "/affinitygroups/%s"
+
+	errParameterNotSpecified = "Parameter %s not specified."
+)
+
+// AffinityGroupClient simply contains a management.Client and has
+// methods for doing all affinity group-related API calls to Azure.
+type AffinityGroupClient struct {
+	mgmtClient management.Client
+}
+
+// NewClient returns an AffinityGroupClient with the given management.Client.
+func NewClient(mgmtClient management.Client) AffinityGroupClient {
+	return AffinityGroupClient{mgmtClient}
+}
+
+// CreateAffinityGroup creates a new affinity group.
+//
+// https://msdn.microsoft.com/en-us/library/azure/gg715317.aspx
+func (c AffinityGroupClient) CreateAffinityGroup(params CreateAffinityGroupParams) (management.OperationID, error) {
```

Comment:
I believe affinity groups are created immediately, we return operationIDs when we kick off a long-running operation, like Create Deployment that returns HTTP 202 Accepted and then expect users to poll on the operation ID for completion.

I don't think that's the case here.